src/kernel/sporadic.c

```diff
@@ -219,8 +219,10 @@ void refill_update(sched_context_t *sc, ticks_t new_period, ticks_t new_budget,
         REFILL_HEAD(sc).rAmount = new_budget;
     } else {
         /* otherwise schedule the rest for the next period */
-        refill_t new = { .rAmount = (new_budget - REFILL_HEAD(sc).rAmount),
-                         .rTime = REFILL_HEAD(sc).rTime + new_period
+        ticks_t unused = new_budget - REFILL_HEAD(sc).rAmount;
```

Comment:
it was never the intention for round robin threads to have sporadic server refill logic applied to them.

IIRC this wasn't called on RR threads. But I could be remembering wrong.

Reply:
That wasn't clear to me. All of the checks for readiness are applied to round robin tasks. I also wanted to make sure that if a round-robin thread was pre-empted it didn't get an entire budget back when it was resumed. If it did it could potentially starve all other threads at the same prio.

Reply:
All of the checks for readiness are applied to round robin tasks. --> yup, to minimise branches
 If it did it could potentially starve all other threads at the same prio. --> this is confusing. it should accumulate budget in the tail refill.

Reply:
@xurtis the semantics should be like so:

- if a round robin thread is preempted and it will be appended to the end of the scheduling queue, it should have it's budget completely refilled
- only if the round robin thread is preempted and is going to be placed at the head of its scheduling queue should it's budget not be completely refilled